apps/speed.c

```diff
@@ -561,52 +639,514 @@ static OPT_PAIR ecdh_choices[] = {
+            break;
+        }
+    }
+    return count;
+}
+
+static int RSA_verify_loop(void *args) {
+    int ret, count;
+    for (count = 0; COND(rsa_c[j][1]); count++) {
+        ret = RSA_verify(NID_md5_sha1, buf, 36, buf2, rsa_num, rsa_key[j]);
+        if (ret <= 0) {
+            BIO_printf(bio_err, "RSA verify failure\n");
+            ERR_print_errors(bio_err);
+            count = -1;
+            break;
+        }
+    }
+    return count;
+}
+#endif
+
+#ifndef OPENSSL_NO_DSA
+static unsigned int kk;
+static DSA *dsa_key[DSA_NUM];
+static long dsa_c[DSA_NUM][2];
+static int DSA_sign_loop(void *args) {
+    int ret, count;
+    for (count = 0; COND(dsa_c[j][0]); count++) {
+        ret = DSA_sign(EVP_PKEY_DSA, buf, 20, buf2, &kk, dsa_key[j]);
+        if (ret == 0) {
+            BIO_printf(bio_err, "DSA sign failure\n");
+            ERR_print_errors(bio_err);
+            count = 1;
+            break;
+        }
+    }
+    return count;
+}
+
+static int DSA_verify_loop(void *args) {
+    int ret, count;
+    for (count = 0; COND(dsa_c[j][1]); count++) {
+        ret = DSA_verify(EVP_PKEY_DSA, buf, 20, buf2, kk, dsa_key[j]);
+        if (ret <= 0) {
+            BIO_printf(bio_err, "DSA verify failure\n");
+            ERR_print_errors(bio_err);
+            count = 1;
+            break;
+        }
+    }
+    return count;
+}
+#endif
+
+#ifndef OPENSSL_NO_EC
+static unsigned char ecdsasig[256];
+static unsigned int ecdsasiglen;
+static EC_KEY *ecdsa[EC_NUM];
+static long ecdsa_c[EC_NUM][2];
+static int ECDSA_sign_loop(void *args) {
+    int ret, count;
+    for (count = 0; COND(ecdsa_c[j][0]); count++) {
+        ret = ECDSA_sign(0, buf, 20,
+                ecdsasig, &ecdsasiglen, ecdsa[j]);
+        if (ret == 0) {
+            BIO_printf(bio_err, "ECDSA sign failure\n");
+            ERR_print_errors(bio_err);
+            count = 1;
+            break;
+        }
+    }
+    return count;
+}
+
+static int ECDSA_verify_loop(void *args) {
+    int ret, count;
+    for (count = 0; COND(ecdsa_c[j][1]); count++) {
+        ret = ECDSA_verify(0, buf, 20, ecdsasig, ecdsasiglen,
+                ecdsa[j]);
+        if (ret != 1) {
+            BIO_printf(bio_err, "ECDSA verify failure\n");
+            ERR_print_errors(bio_err);
+            count = 1;
+            break;
+        }
+    }
+    return count;
+}
+
+static unsigned char secret_a[MAX_ECDH_SIZE], secret_b[MAX_ECDH_SIZE];
+static EC_KEY *ecdh_a[EC_NUM], *ecdh_b[EC_NUM];
+static int outlen;
+static void *(*kdf) (const void *in, size_t inlen, void *out,
+        size_t *xoutlen);
+
+static int ECDH_compute_key_loop(void *args) {
+    int count;
+    for (count = 0; COND(ecdh_c[j][0]); count++) {
+        ECDH_compute_key(secret_a, outlen,
+                EC_KEY_get0_public_key(ecdh_b[j]),
+                ecdh_a[j], kdf);
+    }
+    return count;
+}
+#endif
+
+
+static int run_benchmark(int async_jobs, loopargs **array_loopargs, int (*loop_function)(void *)) {
```

Comment:
I am happy to swap the two, like:

    static int run_benchmark(int async_jobs, int (*loop_function)(void *), loopargs **array_loopargs)